src/ado2gh/Program.cs

```diff
@@ -23,15 +25,41 @@ public static async Task Main(string[] args)
                 .AddSingleton<AdoApiFactory>()
                 .AddSingleton<GithubApiFactory>()
                 .AddSingleton<RetryPolicy>()
+                .AddSingleton<VersionChecker>()
+                .AddSingleton<IVersionProvider, VersionChecker>()
```

Comment:
This will create another singleton instance of `VersionChecker` so we will end up with two which probably not what we want, to fix it we can resolve the singleton instance like this:
```suggestion
                .AddSingleton<IVersionProvider, VersionChecker>(sp => sp.GetRequiredService<VersionChecker>())
```